### PR TITLE
Implement default_canvas_cull_mask

### LINKED ISSUE #11122: Add a Default Canvas Cull Mask project setting
### Describe the project you are working on

2D Shooter game with minimap which uses SubViewport and visibility_layers

### Describe the problem or limitation you are having in your project

Minimap works this way: all its elements are on visibility layer 2, and everything else by default is on layer 1. I have MinimapViewport which have canvas_cull_mask set to 2. And I need to hide all CanvasItems with layer 2 on the root Viewport. To do this, I can write some initialization code, like `get_tree().root.set_canvas_cull_mask_bit(2, false)`, from editor interface I can't do this.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add project setting `rendering/viewport/default_canvas_cull_mask` which sets `canvas_cull_mask` on root Viewport on startup, providing more friendly interface than editing in code.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Just single project setting `rendering/viewport/default_canvas_cull_mask`, which sets `canvas_cull_mask` on root Viewport on startup, but only in Game, not the Editor (I want to see all CanvasItems in the editor regardless of this setting)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can be worked around, but we have so much settings which just set something on root Viewport on startup, so, why not?

### Is there a reason why this should be core and not an add-on in the asset library?

It's core

### PR DESCRIPTION
**Pull request by Summersay415**
Closes https://github.com/godotengine/godot-proposals/issues/11122

### CODE REVIEW

File: core/config/project_settings.cpp [hunk 1]
```diff
@@ -1570,6 +1570,8 @@ ProjectSettings::ProjectSettings() {
 	GLOBAL_DEF_BASIC(PropertyInfo(Variant::INT, "rendering/textures/canvas_textures/default_texture_filter", PROPERTY_HINT_ENUM, "Nearest,Linear,Linear Mipmap,Nearest Mipmap"), 1);
 	GLOBAL_DEF_BASIC(PropertyInfo(Variant::INT, "rendering/textures/canvas_textures/default_texture_repeat", PROPERTY_HINT_ENUM, "Disable,Enable,Mirror"), 0);
 
+	GLOBAL_DEF_BASIC(PropertyInfo(Variant::INT, "rendering/viewport/default_canvas_cull_mask", PROPERTY_HINT_LAYERS_2D_RENDER), Variant(4294967295));
```
Comment (Summersay415) [inline core/config/project_settings.cpp:1573]: Wrapped Int in Variant because CI fails

**Comment by Summersay415** [comment]
What I need to do to fix CI?

File: core/config/project_settings.cpp [hunk 2]
```diff
@@ -1570,6 +1570,8 @@ ProjectSettings::ProjectSettings() {
 	GLOBAL_DEF_BASIC(PropertyInfo(Variant::INT, "rendering/textures/canvas_textures/default_texture_filter", PROPERTY_HINT_ENUM, "Nearest,Linear,Linear Mipmap,Nearest Mipmap"), 1);
 	GLOBAL_DEF_BASIC(PropertyInfo(Variant::INT, "rendering/textures/canvas_textures/default_texture_repeat", PROPERTY_HINT_ENUM, "Disable,Enable,Mirror"), 0);
 
+	GLOBAL_DEF_BASIC(PropertyInfo(Variant::INT, "rendering/viewport/default_canvas_cull_mask", PROPERTY_HINT_LAYERS_2D_RENDER), 4294967295L);
```
Comment (arkology) [inline core/config/project_settings.cpp:1573]:
```suggestion
	GLOBAL_DEF_BASIC(PropertyInfo(Variant::INT, "rendering/viewport/default_canvas_cull_mask", PROPERTY_HINT_LAYERS_2D_RENDER), 4294967295);
```
Maybe this one fails CI? Idk actually.

Comment (Summersay415) [reply core/config/project_settings.cpp:1573]: Without L was the first Variant, and it still fails

**Comment by ydeltastar** [comment]
This should probably go in the `SceneTree`'s [initialization](https://github.com/godotengine/godot/blob/e65a23762b36b564eb94672031f37fdadba72333/scene/main/scene_tree.cpp#L1876C2-L1880C35) instead of the main function.

**Comment by Summersay415** [comment]
> This should probably go in the `SceneTree`'s [initialization](https://github.com/godotengine/godot/blob/e65a23762b36b564eb94672031f37fdadba72333/scene/main/scene_tree.cpp#L1876C2-L1880C35) instead of main.
The problem is that it needs to apply only in game, not in the editor. In main there is a clean check, what should I do in SceneTree.

**Comment by ydeltastar** [comment]
Project settings should be applied in real-time in the editor so you can preview it and match runtime as closely as possible.
You can [override it](https://docs.godotengine.org/en/stable/tutorials/export/feature_tags.html#overriding-project-settings) for a feature tag like `editor_runtime` when you want a setting to apply only in the running project.

**Comment by Summersay415** [comment]
> > The problem is that it needs to apply only in game, not in the editor.
> 
> Project settings should be applied in real-time in the editor so you can preview it and match runtime as closely as possible.
> You can [override it](https://docs.godotengine.org/en/stable/tutorials/export/feature_tags.html#overriding-project-settings) for feature tags like `editor_runtime` and `release` if you want a setting to apply only in the running project.

Good idea!

**Comment by Summersay415** [comment]
> > The problem is that it needs to apply only in game, not in the editor.
> 
> Project settings should be applied in real-time in the editor so you can preview it and match runtime as closely as possible. You can [override it](https://docs.godotengine.org/en/stable/tutorials/export/feature_tags.html#overriding-project-settings) for feature tags like `editor_runtime` and `release` if you want a setting to apply only in the running project.

I looked into this variant and found that it will be very non-user-friendly: to run only in game you will need 3 settings (one for editor, another for editor_runtime and one more without feature tag). So, this way it will be implemented very strangely.

**Pull request closed without merging**